pkg/kubectl/cmd/exec.go

```diff
@@ -195,26 +242,10 @@ func (p *ExecOptions) Run() error {
 	}
 
 	// ensure we can recover the terminal while attached
-	t := term.TTY{
-		Parent: p.InterruptParent,
-		Out:    p.Out,
-	}
-
-	// check for TTY
-	tty := p.TTY
-	if p.Stdin {
-		t.In = p.In
-		if tty && !t.IsTerminalIn() {
-			tty = false
-			fmt.Fprintln(p.Err, "Unable to use a TTY - input is not a terminal or the right kind of file")
-		}
-	} else {
-		p.In = nil
-	}
-	t.Raw = tty
```

Comment:
Note that the logic changed here for `!p.Stdin && p.TTY`. I guess that's intentional, i.e. that `!p.Stdin => !t.Raw` holds now.

Wondering about having an output-only container without the need to have a stdin...